src/h5web/visualizations/shared/models.ts

```diff
@@ -23,19 +23,24 @@ export type Size = { width: number; height: number };
 
 export type Domain = [number, number];
 
-export interface IndexAxisConfig {
-  indexDomain: Domain;
+export type IndexAxisConfig = {
+  domain: Domain;
   showGrid?: boolean;
-  scaleType?: never; // invalid
-}
+  isIndexAxis: true;
+};
 
-export interface DataAxisConfig {
-  dataDomain: Domain;
+export type DataAxisConfig = {
+  domain: Domain;
   showGrid?: boolean;
   scaleType?: ScaleType;
-}
+};
 
-export type AxisConfig = IndexAxisConfig | DataAxisConfig;
+export type AxisConfig = {
+  domain: Domain;
+  showGrid?: boolean;
+  scaleType: ScaleType;
+  isIndexAxis?: boolean;
```

Comment:
You could make this required and add `isIndexAxis?: false` in `DataAxisConfig.

Reply:
It may also make it easier to add `scaleFn` in this interface, and compute it in `AxisSystemProvider`, before injecting the final config.